server/src/main/java/io/envoyproxy/controlplane/server/DiscoveryServerCallbacks.java

```diff
@@ -54,17 +55,42 @@ default void onStreamOpen(long streamId, String typeUrl) {
    *     will be returned to the client and the stream will be closed with error.
    */
   default void onStreamRequest(long streamId, DiscoveryRequest request) {
```

Comment:
I left the old method name the same to minimize impact on end users, but maybe we should force a rename? failing to implement the v3 version and turning on v3 transport in the data plane would cause issues